03-Collections/Collections/Collections.cs

```diff
@@ -125,8 +171,45 @@ public class Task {
         ///   source = { 1,2,3,4 }, count=5 => ArgumentOutOfRangeException
         /// </example>
         public static IEnumerable<T[]> GenerateAllPermutations<T>(T[] source, int count) {
-            // TODO : Implement GenerateAllPermutations method
-            throw new NotImplementedException();
+            if (source.Length < count || count < 0)
+                throw new ArgumentOutOfRangeException();
+            else
+            {
+                List<T[]> result = new List<T[]>();
+                if(count!=0)
+                {
+                    int[] combination = new int[count];
+                    for (int i = 0; i < count; i++)
+                        combination[i] = i;
```

Comment:
add space after loop or condition when not using brackets, will be more readable